packages/react-resize-handle/src/hooks/useResizeHandle.ts

```diff
@@ -83,20 +98,52 @@ export const useResizeHandle = (params: UseResizeHandleParams) => {
       handleRef.current?.setAttribute(key, String(value));
     });
 
-    onChange?.(currentValue.current);
+    // Make sure to only apply the value if it's not the initial value!
+    if (currentValue.current !== UNMEASURED) {
+      wrapperRef.current?.style.setProperty(
+        variableName,
+        `${currentValue.current}px`
+      );
+      onChange?.(currentValue.current);
+    }
   }, [getA11ValueText, maxValue, minValue, onChange, variableName]);
 
   // In case the maxValue or minValue is changed, we need to make sure we are not exceeding the new limits
   React.useEffect(() => {
-    setValue(currentValue.current);
-  }, [maxValue, minValue, updateElementsAttrs]);
+    if (currentValue.current === UNMEASURED) {
+      return;
+    }
+    currentValue.current = clamp(currentValue.current, minValue, maxValue);
+  }, [maxValue, minValue]);
 
   const setValue = React.useCallback(
     (value: number) => {
-      const newValue = clamp(value, minValue, maxValue);
+      const newValue = relative ? value : clamp(value, minValue, maxValue);
+
       if (newValue !== currentValue.current) {
+        // Save the current value, we might need to revert to it if the new value doesn't have any impact on size
+        const oldValue = currentValue.current;
+
+        // Measure the size before setting the new value
+        const previousSizeInPx = elementDimension(
+          elementRef.current,
+          growDirection
+        );
+
+        // Set the new value and update the elements, this should result in element resize
         currentValue.current = newValue;
         updateElementsAttrs();
+
+        // Measure the size after setting the new value
+        const newSizeInPx = elementDimension(elementRef.current, growDirection);
+
+        // If the size hasn't changed, we need to revert to the old value to keep the state and DOM in sync.
+        // If we don't do this, then the handle might be stuck in a place where small changes
+        // in value don't have any effect.
+        if (newSizeInPx === previousSizeInPx) {
```

Comment:
the `newValue` is the newly calculated potential size for the element, based on the mouse movement for example. What happens when the user tries to resize the element above its maximum possible size (lets say the layout only allows it to go 350px, but they move the cursor right 10 px, newValue will be 360px), this would result into us setting the new value larger and the a11y value being wrong (reading 360 instead of 350). 

Previously this would also result in a 10px "buffer" when the user tries to resize again, but I started to actually measure the element every time they go and resize, making sure we always start from the actual value (in the absolute mode). In relative mode, we are also relying on this code here to only update the css value if it actually has any impact on the element size.

Hope this all makes sense :)